src/textures/mesh_attribute.cpp

```
@@ -0,0 +1,206 @@
+            }
+        } else {
+            switch (m_size) {
+                case 1: result = (Object *) new Impl<1, false>(props); break;
+                case 3: result = (Object *) new Impl<3, false>(props); break;
+
+                default:
+                    Log(Error, "Unsupported attribute size count: %d (expected 1 or 3)", m_size);
+                    Throw("Unsupported attribute size count: %d (expected 1 or 3)", m_size);
+            }
+        }
+
+        return { result };
+    }
+
+    MTS_DECLARE_CLASS()
+protected:
+    std::string m_name;
+    std::string m_type;
+    size_t m_size;
+    float m_scale;
+};
+
+template <typename Float, typename Spectrum, uint32_t Size, bool Vertex>
+class MeshAttributeImpl final : public Texture<Float, Spectrum> {
+public:
+    MTS_IMPORT_TYPES(Texture)
+
+    MeshAttributeImpl(const Properties &props)
+    : Texture(props) {
+        m_name = props.string("name");
+        m_scale = props.float_("scale");
+    }
+
+    UnpolarizedSpectrum eval(const SurfaceInteraction3f &si, Mask active) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::TextureEvaluate, active);
+
+        auto result = interpolate(si, active);
+
+        if constexpr (Size == 3 && is_monochromatic_v<Spectrum>)
+            return luminance(result);
+        else
+            return result;
+    }
+    
+    Float eval_1(const SurfaceInteraction3f &si, Mask active = true) const override {
+        MTS_MASKED_FUNCTION(ProfilerPhase::TextureEvaluate, active);
+
+        if constexpr (Size == 3 && is_spectral_v<Spectrum>) {
+            ENOKI_MARK_USED(si);
+            Throw("eval_1(): The vertex attribute %s was queried for a scalar value, but "
+                  "conversion into spectra was requested!",
+                  to_string());
+        } else {
+            auto result = interpolate(si, active);
+
+            if constexpr (Size == 3)
+                return luminance(result);
```

I would just throw an error here